src/main/java/uk/gov/hmcts/reform/sandl/snlevents/mappers/FactsMapper.java

```diff
@@ -127,36 +101,19 @@ public String mapDbSessionToRuleJsonMessage(Session session) {
             factSession.setRoomId(session.getRoom().getId().toString());
         }
 
+        String sessionMsg;
+        List<String> hearingPartsMsg;
         try {
-            return objectMapper.writeValueAsString(factSession);
+            sessionMsg = objectMapper.writeValueAsString(factSession);
+            // its not possible to do session.getHearingParts() as session is often detached
+            hearingPartsMsg = hearingParts.stream()
+                .map(hp -> mapHearingPartToRuleJsonMessage(hp))
+                .collect(Collectors.toList());
         } catch (JsonProcessingException e) {
-            throw new WebServiceException("Cannot map session to ft", e);
-        }
-    }
-
-    // to be removed?
-    @Deprecated
-    public String mapDbHearingToRuleJsonMessage(Hearing hearing) throws JsonProcessingException {
-        FactHearingPart factHearingPart = new FactHearingPart();
-
-        factHearingPart.setDuration(hearing.getDuration());
-        factHearingPart.setCaseTypeCode(hearing.getCaseType().getCode());
-        factHearingPart.setHearingTypeCode(hearing.getHearingType().getCode());
-        factHearingPart.setScheduleStart(hearing.getScheduleStart());
-        factHearingPart.setScheduleEnd(hearing.getScheduleEnd());
-        factHearingPart.setCreatedAt(hearing.getCreatedAt());
-
-        HearingPart hearingPart = hearing.getHearingParts().get(0); // @TODO temporary solution
-        factHearingPart.setId(hearingPart.getId().toString());
-
-        if (hearingPart.getSessionId() != null) {
-            factHearingPart.setSessionId(hearingPart.getSessionId().toString());
+            throw new WebServiceException("Cannot map session or its hearing part to fact", e);
```

Comment:
I know it was WebServiceException earlier but shouldn't we use SnlEventsException?